h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -495,9 +481,13 @@
         if( !firstChunk || bits1 == null ) { // No more data available or allowed
           // If we are mid-parse of something, act like we saw a LF to end the
           // current token.
-          if(c == CHAR_LF || c == CHAR_CR) quoteCount++;
           if ((state != EXPECT_COND_LF) && (state != POSSIBLE_EMPTY_LINE)) {
-            c = CHAR_LF;  continue; // MAIN_LOOP;
+            c = CHAR_LF;
+            if(state == STRING) {
```

Comment:
Once the parser truly reaches the end, just set quoteCount to zero, as it does not matter anymore and set state to STRING_END. Only for String.